src/Columns/ColumnLowCardinality.cpp

```diff
@@ -629,18 +628,27 @@ void ColumnLowCardinality::Index::convertPositions()
             size_t size = data.size();
             auto new_positions = ColumnVector<IndexType>::create(size);
             auto & new_data = new_positions->getData();
-
-            /// TODO: Optimize with SSE?
-            for (size_t i = 0; i < size; ++i)
+            size_t i=0;
+
+#if defined(__SSE2__)
+
+            const auto bytes_sse = sizeof(__m128i);
+            const __m128i * sse_data_pos = reinterpret_cast<const __m128i *>(&data);
+            const  auto loop_count_sse = (sizeof(IndexType)*size)/bytes_sse;
+            __m128i * sse_new_data_pos = reinterpret_cast<__m128i *>(&new_data);
+            for (size_t j=0;j<loop_count_sse;j++,sse_data_pos += bytes_sse,sse_new_data_pos += bytes_sse)
+                _mm_storeu_si128(sse_new_data_pos , _mm_loadu_si128(sse_data_pos));
```

Comment:
This code is wrong because the types of `new_data` and `data` are different.
If they were the same, it would be enough to use `memcpy`.